### ISSUE TITLE
Aw Snap errors

### ISSUE THREAD

**Issue by captainjdi** [FlutterFlow/flutterflow-issues#1747]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

While working in FlutterFlow, my project keeps crashing the browser and showing an "Aw snap" error with "Error code: 5." There aren't specific steps to reproduce the issue. FlutterFlow breaks at random times while I'm working in it, usually while editing the UI of screens for 15-20 minutes. This happens over and and over again throughout the day. I contacted customer support, and after a week of back and forth they directed me to file a ticket here. Here is my project: https://app.flutterflow.io/project/olympus-money-dfc8ai?tab=uiBuilder&page=Overview

I have narrowed down the scope of what could be causing the issue. I created a few copies of my project and began removing different parts of it to isolate the root cause. While doing so, I found that the "Aw snap" errors went away when I removed all the Actions from the Action Flow Editor on the pages in my Onboarding folder (that's where most of the Actions in my project exist so far). I've added very robust actions that call custom functions that require dozens of parameters. My guess is that something with that could be the issue, but I'm not doing anything wrong. I'm just doing a lot. 

So, it appears as though something I'm doing with my Actions is causing FlutterFlow to struggle and break. This manifests itself as the "Aw snap" error. 

I was instructed to create this ticket by customer support: 

Hi JD,

Thank you so much for taking the time to prepare this information, this was incredibly helpful!
​﻿
As we cant reproduce the issue, As a next step, could you please [head to the GitHub Issue tracker and report this bug](https://github.com/FlutterFlow/flutterflow-issues/issues/new?assignees=&labels=status%3A+needs+triage&template=bug_report.yml)? The Development Team will likely have some follow up questions, so it would be best if you created this issue. You know how your project works better than anyone!

Once you've created this issue, could you please share the Issue # with me. That way I can follow along as the team works through this.

Thanks for your support.

Best,

Sam

### Expected Behavior

I do not expect the browser to crash with an "Aw snap" error every 15-20 minutes while using the app. 

### Steps to Reproduce

1. Open the project I'm working on
2. Go to a UI screen and make changes for 15-20 minutes
3. An "Aw snap" error will be triggered
4. Note: I'm going to mark the checkbox below even though this is not STR from a blank project. Again, I was told by customer success to file this ticket and need your help. I've done a lot of legwork in identifying the area of the app that is not working and need technical support. Uncovering this issue should help future users not run into the same limitations that I'm hitting. 

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT40icn65YtPvttK+KqJb8FvmmU5JlgiUK8Viu57aCgsfaLuBJkQdvWlRxBVZ73gSlFUOkSmmn4z+cmKhvDpGvY7NSeCGYQ5+c5ySTnLRmKiQZOBDMyJPnJCHcdhCUeM0J6ruiQmLvNedU0s72eHftyWbBbfQIO2IndISq/LZO4=

### Context

I can't work for more than 15 minutes without the app breaking. Every time it breaks I lose work. I can't confidently rely on the app to build what I need to build and feel like I may have wasted months of work. I'm in a full-blown panic because I'm not sure the product can support the performance I need. Please, please help and have empathy for me as a customer and other customers who will also run into this issue. 

### Visual documentation

<img width="1279" alt="image" src="https://github.com/FlutterFlow/flutterflow-issues/assets/133725139/a2772162-8d81-4656-b0c8-508f4751e3da">

### Additional Info

https://drive.google.com/file/d/1RzoAeTOM_CPDswMBx10nMuklDSv6sbjV/view?usp=sharing

### Environment

```markdown
- FlutterFlow version: v4.0 Flutter 3.13.7
- Platform: ?
- Browser name and version: Chrome Version 118.0.5993.117 (Official Build) (x86_64). Also happens in Firefox and Safari. 
- Operating system and version affected: macOS Monterey version 12.6.7
```

**Comment by katherineqian** [comment]
Hi @captainjdi, sorry for the late response. I've been investigating the issue with your project. While I can't reproduce the issue on my end, based on profiling the performance, I believe the performance issues you're experiencing could be caused by the large number of App State fields in your project as well as the related update local state actions (likely used in the onboarding folder, which is why the issues resolved when removing those actions). In general, we do not recommend using this many fields in App State, especially to store constant fields like colors or container dimensions. We'll be releasing a feature that should help you migrate some of your fields off of App State soon - I can't share more than that, but I think it's coming out with the release this Thursday or next Thursday. 

Otherwise, I can't determine if there is a different reproducible error with this project. I'll keep this open, and if you are another user experiencing this problem, please thumbs up the issue and leave a comment below with a bug report code.

**Comment by katherineqian** [comment]
Closing for now due to lack of additional reports.

**Comment by captainjdi** [comment]
Hi Kathy,

Apologies, I never saw your response! Overall, I was able to resolve most
of the issues by moving things to Action Blocks. Did you already release
the feature you were referring to in the email? Is it the Constants area?

Is there an easy way to transfer things over to constants from app state
variables, or is that all manual? Thank you SOO much for replying to my
issue. I just signed up for the Pro plan and I'm getting close to releasing
something. Thanks again!

Best,

JD
